lib/mauth/client.rb

```diff
@@ -265,19 +282,40 @@ def signed(object, attributes = {})
       end
 
       # takes a signable object (outgoing request or response). returns a hash of headers to be
-      # applied tothe object which comprise its signature.
+      # applied to the object which comprise its signature.
       def signed_headers(object, attributes = {})
+        v1_only_override = attributes.delete(:v1_only_override)
+        v2_only_override = attributes.delete(:v2_only_override)
+
+        if v1_only_override # used when signing responses to requests with only the v1 protocol, when AUTHENTICATE_WITH_ONLY_V2 is false and SIGN_REQUESTS_WITH_ONLY_V2 is true
+          signed_headers_v1(object, attributes)
+        elsif sign_requests_with_only_v2? || v2_only_override # override used when signing responses to requests with the v2 protocol when SIGN_REQUESTS_WITH_ONLY_V2 is false
+          signed_headers_v2(object, attributes)
+        else # by default sign with both the v1 and v2 protocol
+          signed_headers_v1(object, attributes).merge(signed_headers_v2(object, attributes))
+        end
+      end
+
+      def signed_headers_v1(object, attributes = {})
         attributes = { time: Time.now.to_i.to_s, app_uuid: client_app_uuid }.merge(attributes)
-        signature = self.signature(object, attributes)
+        string_to_sign = object.string_to_sign_v1(attributes)
+        signature = self.signature(string_to_sign)
         { 'X-MWS-Authentication' => "#{MWS_TOKEN} #{client_app_uuid}:#{signature}", 'X-MWS-Time' => attributes[:time] }
       end
 
-      # takes a signable object (outgoing request or response). returns a mauth signature string
-      # for that object.
-      def signature(object, attributes = {})
-        assert_private_key(UnableToSignError.new("mAuth client cannot sign without a private key!"))
+      def signed_headers_v2(object, attributes = {})
```

Comment:
I insist v2 code should be separated. Just this diff here is quite confusing, did signature method disappear?  Etc.
This client file is a mess already, adding more to it makes it really complex to follow

Reply:
@jcarres-mdsol was trying to get some clarity on how to do that here: https://github.com/mdsol/mauth-client-ruby/pull/25#discussion_r292625902, but just moved v2 only methods from `client.rb` into their own file.